internal/action/actions.go

```diff
@@ -160,13 +160,19 @@ func (h *BufPane) Center() bool {
 	return true
 }
 
-// MoveCursorToView moves the cursor to the current view
-func (h *BufPane) MoveCursorToView() bool {
+// CursorToView moves the cursor to the current view
+func (h *BufPane) CursorToView() bool {
 	v := h.GetView()
-	h.Cursor.GotoLoc(buffer.Loc{0, v.StartLine.Line})
-	h.RemoveAllMultiCursors()
 	h.Cursor.Deselect(true)
-	h.Relocate()
+	loc := h.Cursor.Loc
+	if(loc.Y < v.StartLine.Line || loc.Y > v.StartLine.Line + h.BufView().Height - 1) {
+		scrollmargin := int(h.Buf.Settings["scrollmargin"].(float64))
+		h.Cursor.GotoLoc(h.LocFromVLoc(display.VLoc{
+			SLoc: h.Scroll(v.StartLine, scrollmargin),
+			VisualX: 0,
+		}))
+		h.RemoveAllMultiCursors()
+	}
 	return true
```

Comment:
It would be useful to return false if the cursor is already in the current view, i.e. if `CursorToView` does nothing. We do that in various other cases when the action did nothing because there was nothing to do, e.g. `Deselect` returns false if there is no selection, `NextTab` returns false if there is no next tab etc. That allow conditional chaining of actions with `|` or `&`.

Reply:
Good point.